source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,313 @@
+feature of MongoDB and other databases that lets you define an arbitrary
+sequence of database commands as an atomic operation.  
+
+For more information on atomic operations and atomicity, see 
+:manual:`the MongoDB manual entry for atomicity and transactions </core/write-operations-atomicity/>`.
+
+For more information on transactions, 
+:manual:`see the MongoDB manual entry for transactions </core/transactions/>`.
+
+How to Use Compound Operations
+------------------------------
+
+This section shows how to use each compound operation with the MongoDB Java Driver.
+
+The following examples use a collection containing these two sample documents.
+
+.. code-block:: json
+
+    {"_id": 1, "food": "donut", "color": "green"}
+    {"_id": 2, "food": "pear", "color": "yellow"}
+
+`The full code for the following examples is available on Github here. <https://github.com/mongodb/docs-java/blob/master/source/includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java>`__
+
+.. note:: Before or After the Write?
+
+   By default, compound operations return your found document in the state
+   before your write operation. You can retrieve your found document in the
+   state after your write operation by using the options class corresponding to
+   your compound operation. You can see an example of this configuration in the 
+   :ref:`Find and Replace example below <java-find-and-replace-example>`.
+
+Find and Update
+~~~~~~~~~~~~~~~
+
+To find and update one document, use the ``findOneAndUpdate()`` method of the
+``MongoCollection`` class. The ``findOneAndUpdate()`` method returns your found
+document or ``null`` if no documents match your query.
+
+Example
+^^^^^^^
+
+The following example uses the ``findOneAndUpdate()`` method to find a
+document with the ``color`` field set to ``"green"`` and  update the
+``food`` field in that document to ``"pizza"``. 
+
+The example also uses a ``FindOneAndUpdateOptions`` instance to specify the
+following options:
+
+- Exclude the ``_id`` field from the found document.
+- Specify to insert a document if no documents match the query. MongoDB and
+  other databases refer to this operation as an upsert.
+- Set a maximum execution time of 5 seconds for this operation on the MongoDB
+  instance. If the operation takes longer, the ``findOneAndUpdate()`` method
+  will throw a ``MongoExecutionTimeoutException``.  
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java
+   :language: java
+   :dedent:
+   :start-after: start findOneAndUpdate-example
+   :end-before: end findOneAndUpdate-example
+
+The output of the above code should look like this:
+
+.. code-block:: json
+
+   {"food": "donut", "color": "green"}
+
+For more information on the ``Projections`` class, see our
+:doc:`guide on the Projections builder </fundamentals/builders/projections/>`.
+
+For more information on the methods and classes in this section, see the
```

Comment:
```suggestion
For more information on the method and classes in this section, see the
```
Nit: I noticed this in my own PR too, but "methods and classes" are in it's plural form when "methods" should be in singular.

Applies to all instances.

Reply:
Thanks for the note. I do not believe this edit benefits the page. As the linked method and classes will link to multiple methods and classes (the [`FindOneAndUpdateOptions`](https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/FindOneAndUpdateOptions.html) presents the reader with a list of methods 4 of which are used in the example), I feel changing "methods" to "method" will not improve accuracy (and is sound slightly off to my ear).